multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -774,8 +743,11 @@ def fit(
             label_column=self._label_column,
             column_types=column_types,
             data=train_data,
-            provided_problem_type=self._problem_type,
+            provided_problem_type=self._problem_type.name,
         )
+        if problem_type is not None:
+            problem_type = problem_type_reg.get(problem_type)
+            self._problem_type = problem_type  # In case problem type isn't provided in __init__().
```

Comment:
This logic seems not to work for matching problems. For example, if `self._problem_type.name` is `image_similarity`, the inferred problem types from `infer_problem_type_output_shape ` is a traditional type, e.g., binary. Then `problem_type = problem_type_reg.get(problem_type)` would return the `binary` problem type, which changes the original `image_similarity`.

Reply:
I guess that few shot image classification will also have a similar issue.